src/grade_recognition_systems.asm

```diff
@@ -0,0 +1,383 @@
+; DMGTRIS
+; Copyright (C) 2023 - Randy Thiemann <randy.thiemann@gmail.com>
+
+; This program is free software: you can redistribute it and/or modify
+; it under the terms of the GNU General Public License as published by
+; the Free Software Foundation, either version 3 of the License, or
+; (at your option) any later version.
+
+; This program is distributed in the hope that it will be useful,
+; but WITHOUT ANY WARRANTY; without even the implied warranty of
+; MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+; GNU General Public License for more details.
+
+; You should have received a copy of the GNU General Public License
+; along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+
+IF !DEF(GRADING_SYSTEMS_ASM)
+DEF GRADING_SYSTEMS_ASM EQU 1
+
+
+INCLUDE "globals.asm"
+
+
+SECTION "Grade Variables", WRAM0
+wDecayRate:       ds 1
+wGradePoints:     ds 1
+wInternalGrade:   ds 1
+wDisplayedGrade:: ds 1
+wPreviousGrade:   ds 1
+wEffectTimer::    ds 1
+wPalette::        ds 1
+
+
+SECTION "Grading Data", ROM0
+; The Score Thresholds are 1/4th of the original ones
+sGradeScores:

```

Comment:
Rename to sTGM1GradeScores so that it's obvious to which mode which data applies.